clang/test/dpct/helper_files_ref/include/asm_utils.hpp

```diff
@@ -0,0 +1,38 @@
+//==---- asm.hpp -------------------------------*- C++ -*----------------==//
+//
+// Copyright (C) Intel Corporation
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __DPCT_ASM_UTILS_HPP__
+#define __DPCT_ASM_UTILS_HPP__
+
+#include <sycl/sycl.hpp>
+
+namespace dpct {
+
+inline uint32_t logical_op_3inputs(uint32_t a, uint32_t b, uint32_t c, uint8_t OpTypes) {
+  uint32_t d = 0;
+  if (OpTypes & 0x01)
+    d |= ~a & ~b & ~c;
```

Comment:
Suggest to remove the if statement,  as the branch code if not effiecent on GPU when execution.

Reply:
move this logic out of help function.